### PR TITLE
Towards a more robust autovacuum

### PR DESCRIPTION
**Pull request by bouvyd**
Imagine you have 2 transient models, Wizard and WizardLine, with a
m2o field set to required on the WizardLine to the Wizard model.

On autovacuuming, the autovacuum will crash on the table for the
Wizard model since WizardLine rows still references it with a required
m2o, which by default behaves like an ondelete='restrict' field.

Before this revision, the autovacuuming stopped there. Then it would
try to do the same thing every day, crashing each time. This also meant
that no table would, in fact, get vacuumed at all since the transaction
was rollbacked (no auto-commit).

After this revision, the autovacuuming will crash & skip the Wizard table,
proceed to the WizardLine table and delete old rows; on the next day
the Wizard table should be cleaned as well since it is no longer
referenced.

Note that the right thing to do would be to either explicitly add
ondelete='cascade' attributes on relational fields of TransientModels
or simply overwrite the base TransientModel class to assume all m2o
fields should behave that way. Still, this simple precaution could
avoid a lot of wasted rows.

TL;DR: crashes of autovacuuming only impact one table and not the
whole DB.

Note: [FIX] or [IMP]? not sure...

### CODE REVIEW

**Comment by dbo-odoo** [comment]
@rco-odoo 
Could you have a look at this when you have some time?

**Comment by dbo-odoo** [comment]
@rco-odoo  up!
